packages/core/admin/ee/server/register.js

```diff
@@ -1,16 +1,14 @@
 'use strict';
 
-const { features } = require('@strapi/strapi/lib/utils/ee');
+const features = require('./utils/features');
 const executeCERegister = require('../../server/register');
 const migrateAuditLogsTable = require('./migrations/audit-logs-table');
 const createAuditLogsService = require('./services/audit-logs');
 const reviewWorkflowsMiddlewares = require('./middlewares/review-workflows');
 const { getService } = require('./utils');
 
 module.exports = async ({ strapi }) => {
-  const auditLogsIsEnabled = strapi.config.get('admin.auditLogs.enabled', true);
```

Comment:
Yes it was intended :) 

It's because of the way we handle license changes at runtime (see #15913). Basically we always register the service (unless the user disabled it), and inside that service we have the logic to check whether the license allows the feature. This way we have a place to listen to all the license events